### PR TITLE
Release 2023.8.0

### LINKED ISSUE #64: Drop `c_compiler_version` from build config
### Comment:

Don't think we have need for a specific C compiler version on any platform, so we should be good to remove the version specification on the next release since it creates some issues on Windows builds

### LINKED ISSUE #66: Remove run-constraint on glibc>=2.32 on aarch
AFAIU the TLS allocation situation, it's not _absolutely_ required to use newer glibc, i.e. things can still be "solved" by ensuring the libraries that require static TLS allocations come first (e.g. by reordering imports), see also: https://github.com/conda-forge/conda-forge.github.io/issues/1551

In particular, glibc 2.32 is very aggressive, and is only supported by around ~15% of Python users ([details](https://mayeut.github.io/manylinux-timeline/)[^1]). It also switches off the testing here, which is IMO worse than living with these TLS allocation problems (because we might end up pushing packages that seem to build but are broken).

CC @conda-forge/dask-sql @jakirkham 

[^1]: see the plot "glibc version (across non-EOL Python versions)"  for the consumer-side statistics. This only shows percentages for >=2.31 and >=2.34, but looking at https://github.com/mayeut/pep600_compliance/, there are no major distros using glibc 2.32 & 2.33 that aren't already EOL, so I think the percentage for >=2.34 will apply almost unchanged for >=2.32.

### LINKED ISSUE #1169: ARM GLIBC pre-2.32 TLS issue
Currently the `aarch64` builds require `__glibc >=2.32` to prevent TLS allocation error when importing; is there any way we could modify the TLS model of its shared object to avoid this?

Filing this issue to track

ref: https://bugzilla.redhat.com/show_bug.cgi?id=1722181

### PR DESCRIPTION
**Pull request by ayushdg**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/dask-sql-feedstock/issues/64
Fixes https://github.com/conda-forge/dask-sql-feedstock/issues/66
Fixes https://github.com/dask-contrib/dask-sql/issues/1169

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Ayush! 🙏

**Comment by jakirkham** [comment]
Could you please drop these lines and [re-render]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks )?

https://github.com/conda-forge/dask-sql-feedstock/blob/4ec2ca4dbe6b95fcaa20e9bb20184d4fca1f6824/recipe/conda_build_config.yaml#L1-L3

This will resolve issue ( https://github.com/conda-forge/dask-sql-feedstock/issues/64 )

**Comment by ayushdg** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,34 +29,27 @@ requirements:
...
     - libprotobuf
+    - xz  # [linux64]
   run:
     - python
-    - dask >=2022.3.0,<=2023.5.1
+    - dask >=2022.3.0,<=2023.7.1
     - pandas >=1.4.0
-    # FIXME: handling is needed for httpx-based fastapi>=0.87.0
-    - fastapi >=0.69.0,<0.87.0
+    - fastapi >=0.92.0
```
Comment (jakirkham) [inline recipe/meta.yaml:37]: Am seeing [`httpx` added upstream]( https://github.com/dask-contrib/dask-sql/compare/2023.6.0...2023.8.0#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R49 ). Should we add it here?

```suggestion
    - fastapi >=0.92.0
    - httpx >=0.24.1
```

Comment (ayushdg) [reply recipe/meta.yaml:37]: Nice catch! I ended up missing this despite using grayskull to generate a meta.yaml I could look at.

Comment (jakirkham) [reply recipe/meta.yaml:37]: Oh interesting. Did `grayskull` not pick this up? If so, think we should [file a bug report]( https://github.com/conda/grayskull/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=%5BBUG%5D+Title+of+the+bug )

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Ayush! 🙏

Added `automerge` so this merges once CI passes

**Pull request merged**